packages/flutter_tools/lib/src/ios/fallback_discovery.dart

```diff
@@ -82,50 +82,50 @@ class FallbackDiscovery {
     }
 
     try {
-      final Uri result = await _mDnsObservatoryDiscovery.getObservatoryUri(
-        packageId,
-        device,
-        usesIpv6: usesIpv6,
-        hostVmservicePort: hostVmservicePort,
-      );
+      final Uri result = await _protocolDiscovery.uri;
       if (result != null) {
         UsageEvent(
           _kEventName,
-          'mdns-success',
+          'log-success',
           flutterUsage: _flutterUsage,
         ).send();
         return result;
       }
+    } on ArgumentError {
+      // In the event of an invalid InternetAddress, this code attempts to catch
+      // an ArgumentError from protocol_discovery.dart
     } on Exception catch (err) {
       _logger.printTrace(err.toString());
     }
-    _logger.printTrace('Failed to connect with mDNS, falling back to log scanning');
+    _logger.printTrace('Failed to connect with log scanning, falling back to mDNS');
     UsageEvent(
       _kEventName,
-      'mdns-failure',
+      'log-failure',
       flutterUsage: _flutterUsage,
     ).send();
 
     try {
-      final Uri result = await _protocolDiscovery.uri;
+      final Uri result = await _mDnsObservatoryDiscovery.getObservatoryUri(
+        packageId,
+        device,
+        usesIpv6: usesIpv6,
+        hostVmservicePort: hostVmservicePort,
+      );
       if (result != null) {
         UsageEvent(
           _kEventName,
-          'fallback-success',
+          'mdns-success',
           flutterUsage: _flutterUsage,
         ).send();
         return result;
       }
-    } on ArgumentError {
```

Comment:
Can this not happen anymore?

Reply:
This change is swapping the order fallbacks.  Before it was trying mDNS and then log scanning, but now it's trying log scanning first.  The reordering confused the diff, this `ArgumentError` was just moved.

Reply:
It's https://github.com/flutter/flutter/pull/66092/commits/e91e25e01ffa596f7e60c861b435a9c672a9a29e#diff-e0fb4f81676c3d33cb1796b5f126a4fdR94 now.